src/nvim/syntax.c

```diff
@@ -5751,185 +5751,96 @@ static void syntime_report(void)
 *  Highlighting stuff		      *
 **************************************/
 
-/*
- * The default highlight groups.  These are compiled-in for fast startup and
- * they still work when the runtime files can't be found.
- * When making changes here, also change runtime/colors/default.vim!
- * The #ifdefs are needed to reduce the amount of static data.  Helps to make
- * the 16 bit DOS (museum) version compile.
- */
-# define CENT(a, b) b
-static char *(highlight_init_both[]) =
-{
-  CENT(
-      "ErrorMsg term=standout ctermbg=DarkRed ctermfg=White",
-      "ErrorMsg term=standout ctermbg=DarkRed ctermfg=White guibg=Red guifg=White"),
-  CENT("IncSearch term=reverse cterm=reverse",
-      "IncSearch term=reverse cterm=reverse gui=reverse"),
-  CENT("ModeMsg term=bold cterm=bold",
-      "ModeMsg term=bold cterm=bold gui=bold"),
-  CENT("NonText term=bold ctermfg=Blue",
-      "NonText term=bold ctermfg=Blue gui=bold guifg=Blue"),
-  CENT("StatusLine term=reverse,bold cterm=reverse,bold",
-      "StatusLine term=reverse,bold cterm=reverse,bold gui=reverse,bold"),
-  CENT("StatusLineNC term=reverse cterm=reverse",
-      "StatusLineNC term=reverse cterm=reverse gui=reverse"),
+// The default highlight groups.  These are compiled-in for fast startup and
+// they still work when the runtime files can't be found.
+//
+// When making changes here, also change runtime/colors/default.vim!
+
+static char *highlight_init_both[] =
+{
+  "DiffText     cterm=bold ctermbg=Red gui=bold guibg=Red",
```

Comment:
If I understand `cterm/gui` not mutally exclusive` correctly, it means that both still work as expected, but there's no need to set both to the same value, since neovim will choose the proper flag depending on the used UI anyway.

cterbg/ctermfg are still needed.. I have to set color somehow. :]